packages/camera/camera/example/ios/RunnerTests/AvailableCamerasTest.m

```diff
@@ -0,0 +1,137 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera;
+@import camera.Test;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+#import "MockFLTThreadSafeFlutterResult.h"
+
+@interface AVCaptureDeviceMock : AVCaptureDevice
```

Comment:
can we just use OCMock for this?

Reply:
well, initially I wanted to use OCMock for that, however, when I wrote:
```
id deviceMock = OCMClassMock([AVCaptureDevice class]);
OCMStub([deviceMock uniqueID]).andReturn(@"1");
```
I was getting: `Multiple methods named 'uniqueID' found with mismatched result, parameter type or attributes` and the same for `position`, so I've decided to go with mock class instead. (sorry if I'm making some obvious mistake, it's my first time with obj-c. :D)

Reply:
You should not use `id` for the variable type. `id` is roughly the equivalent of `Object` in Dart, which means the compiler has absolutely no idea what object to check methods against.

Reply:
ah, cool, now it makes sense. I changed it to use the `OCMClassMock` thx for the explanation